app/controllers/user_controller.rb

```diff
@@ -0,0 +1,44 @@
+module TheApp
+
+  module Routes
+
+    class UserController < Base
+      PERMITTED_PARAMS = [:email, :username, :name, :password, :password_confirmation]
+
+      before do
+        params.each do |k, v|
+          self.class.send(:define_method, k){v} if PERMITTED_PARAMS.include?(k.to_sym)
+          # instance_variable_set("@#{k}", v)
+        end
+      end
+
+      get '/users/new' do
+        erb :'/users/new'
+      end
+
+      post '/users/new' do
+        @user = create_user(params)
+        if @user.save
+          session[:user_id] = @user.id
+          redirect to('/peeps')
+        else
+          flash.now[:errors] = @user.errors.full_messages
+          erb :'/users/new'
+        end
+      end
+
+      def create_user(params)
```

Comment:
Unused method argument - `params`. If it's necessary, use `_` or `_params` as an argument name to indicate that it won't be used. You can also write as `create_user(*)` if you want the method to accept any arguments but don't care about them.